core/constraints/constraints.go

```diff
@@ -39,6 +39,18 @@ const (
 	ImageID          = "image-id"
 )
 
+// CharmValue describes a user's requirements of the hardware on which units
+// of an application will run. Constraints are used to choose an existing machine
+// onto which a unit will be deployed, or to provision a new machine if no
+// existing one satisfies the requirements.
+type CharmValue struct {
+	MemRequest *uint64 `json:"mem-request,omitempty" yaml:"cpu-power,omitempty"`
```

Comment:
No need for tags here. The rpc params struct is what defines the tags.
The only reason the constraints Value has tags is for formatting from the CLI (and that's a bit wrong anyway).
Also, we're always going to set both of these so no need for pointers.